pkg/cloud/baremetal/actuators/machine/actuator_test.go

```diff
@@ -1630,3 +1632,203 @@ func TestDeleteOfBareMetalHostDeletesMachine(t *testing.T) {
+	})
+	if err != nil {
+		t.Error(err)
+	}
+
+	c.Create(context.TODO(), machine)
+	c.Create(context.TODO(), host)
+	c.Create(context.TODO(), node)
+
+	err = actuator.Update(context.TODO(), nil, machine)
+	if err != nil {
+		t.Errorf("unexpected error %v", err)
+	}
+
+	host = &bmh.BareMetalHost{}
+	c.Get(context.TODO(), hostNamespacedName, host)
+	if !hasRebootAnnotation(host) {
+		t.Log("Expected reboot annotation on the host but none found")
+		t.Fail()
+	}
+
+	host.Status.PoweredOn = false
+	c.Update(context.TODO(), host)
+
+	err = actuator.Update(context.TODO(), nil, machine)
+	if err != nil {
+		t.Errorf("unexpected error %v", err)
+	}
+
+	machine = &machinev1.Machine{}
+	c.Get(context.TODO(), machineNamespacedName, machine)
+
+	if _, exists := machine.Annotations[remediationInProgressAnnotation]; !exists {
+		t.Log("Expected remediation-in-progress annotation to exist on machine but none found")
+		t.Fail()
+	}
+
+	err = actuator.Update(context.TODO(), nil, machine)
+	if err != nil {
+		t.Errorf("unexpected error %v", err)
+	}
+
+	node = &corev1.Node{}
+	err = c.Get(context.TODO(), nodeNamespacedName, node)
+	if !errors.IsNotFound(err) {
+		t.Log("Expected node to be deleted")
+		t.Fail()
+	}
+
+	err = actuator.Update(context.TODO(), nil, machine)
+	if err != nil {
+		t.Errorf("unexpected error %v", err)
+	}
+
+	host = &bmh.BareMetalHost{}
+	c.Get(context.TODO(), hostNamespacedName, host)
+
+	if hasRebootAnnotation(host) {
+		t.Log("Expected reboot annotation to be removed but it's still there")
+		t.Fail()
+	}
+
+	//simulate power on and node that comes up again
+	host.Status.PoweredOn = true
+	c.Update(context.TODO(), host)
+
+	node, _ = geteNode()
+	linkMachineAndNode(machine, node)
+	c.Create(context.TODO(), node)
+	c.Update(context.TODO(), machine)
+
+	err = actuator.Update(context.TODO(), nil, machine)
+	if err != nil {
+		t.Errorf("unexpected error %v", err)
+	}
+
+	machine = &machinev1.Machine{}
+	c.Get(context.TODO(), machineNamespacedName, machine)
+
+	if len(machine.Annotations) > 0 {
+		if _, exists := machine.Annotations[externalRemediationAnnotation]; exists {
+			t.Log("Expected external remediation annotation to be removed but it's still there")
+			t.Fail()
+		}
+
+		if _, exists := machine.Annotations[remediationInProgressAnnotation]; exists {
+			t.Log("Expected remediation-in-progress annotation to be removed but it's still there")
+			t.Fail()
+		}
+	}
+
+	//make sure nothing happens after remediation
+	err = actuator.Update(context.TODO(), nil, machine)
+	if err != nil {
+		t.Errorf("unexpected error %v", err)
+	}
+	err = actuator.Update(context.TODO(), nil, machine)
+	if err != nil {
+		t.Errorf("unexpected error %v", err)
+	}
+
+	host = &bmh.BareMetalHost{}
+	c.Get(context.TODO(), hostNamespacedName, host)
+
+	if hasRebootAnnotation(host) {
+		t.Log("Expected reboot annotation to be removed but it's still there, maybe reboot loops?")
+		t.Fail()
+	}
+}
+
+func geteNode() (*corev1.Node, types.NamespacedName) {
```

Comment:
`gete`?

Reply:
The new `getX()` functions might be even more useful if they took a string as a name argument.